module/VuFind/src/VuFind/View/Helper/Root/Icon.php

```diff
@@ -0,0 +1,219 @@
+<?php
+/**
+ * Icon view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+use Laminas\View\Helper\EscapeHtmlAttr;
+use Laminas\View\Helper\HeadLink;
+use VuFind\Cache\Manager as CacheManager;
+use VuFind\Cache\StorageInterface;
+use VuFindTheme\ThemeInfo;
+
+/**
+ * Icon view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Icon extends AbstractHelper
+{
+    /**
+     * Icon config from theme.config.php
+     *
+     * @var array
+     */
+    protected $config;
+
+    /**
+     * Default icon set
+     *
+     * @var string
+     */
+    protected $defaultSet;
+
+    /**
+     * Transforming map
+     *
+     * @var array
+     */
+    protected $iconMap;
+
+    /**
+     * Cache for icons
+     *
+     * @var StorageInterface
+     */
+    protected $cache;
+
+    /**
+     * Escape helper
+     *
+     * @var EscapeHtmlAttr
+     */
+    protected $esc;
+
+    /**
+     * HeadLink helper
+     *
+     * @var HeadLink
+     */
+    protected $headLink;
+
+    /**
+     * Prevent extra work by only appending the stylesheet once
+     *
+     * @var boolean
+     */
+    protected $styleAppended = false;
+
+    /**
+     * Constructor
+     *
+     * @param ThemeInfo      $themeInfo    Theme info helper
```

Comment:
Since we are only using ThemeInfo to fetch configuration, I don't think we should make it a dependency of this class. Instead, we should fetch the configuration in the factory, and pass it here. That will make it easier to write a unit test, and will make things generally more flexible.